### DISCUSSION TITLE
Добавить значение из другой таблицы

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#355] [Как это сделать?] [answered]
Как возможно записать значение в поле одной таблицы из поля другой таблицы, при условии что есть поле ключ связывающий эти таблицы?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
https://ru.docs.totum.online/action-codes-base-level

**Reply by vldekov** [reply]
смотреть setListExtended ?

**Reply by ttm-alexeypolunin** [reply]
Если нужно изменить сразу несколько строк на значения зависящие от строки привязки то да. 

В field список со значениями в where список с соответсвиями

Если взяли через rowList и берете колонки, то строку с rowList обязательно закрепить ~ тк порядок при повторном вызове может изменится.

**Comment by vldekov** [comment]
Получился следующий код:
=:setListExtended(table: $#ntn; field: 'insert' = $sl1; where: 'nomerpaz'=$sl2; log: true)
sl1:selectList(table: 'protokol_analiza_zayavki'; field: 'kontragentimya')
sl2:selectList(table: 'protokol_analiza_zayavki'; field: 'nomerpaz')
Результат:
![image](https://user-images.githubusercontent.com/114401808/234172014-d665d960-becf-46b0-b05a-5f49db628139.png)

**Reply by vldekov** [reply]
Необходимо значение поля kontragentimya ( это ID селекта ) преобразовать в kontragentimya ( имя контрагента ).
=:setListExtended(table: $#ntn; field: 'insert' = $sl3; where: 'nomerpaz'=$sl2; log: true)
sl1:selectList(table: 'protokol_analiza_zayavki'; field: 'kontragentimya')
sl2:selectList(table: 'protokol_analiza_zayavki'; field: 'nomerpaz')
sl3:selectList(table: 'spisokkontragentov'; field: 'kontragentimya'; where: 'id' = $sl1)
Но выдает ОШИБКУ:
В параметре where необходимо использовать лист по количеству изменяемых строк либо не лист.
Функция setListExtended; Линия кода d1.; Функция while; action поле h_test таблицы ttnakabel

Подскажите как возможно это сделать?

**Comment by poluninatm** [comment]
У вас в $sl3 и $sl2 получаются списки с разным количеством значений. Для
setListExtended это критически важно.

Вы приравниваете ид таблицы «список контрагентов» к ид таблицы «протокол
анализа заявки» в строке sl3.

Они точно совпадают?

Так же не задан order для сопоставляемых в setListExtended значений. Если
бы все сработало, данные в изменяемом поле изменились бы случайным образом.

Вт, 25 апр. 2023 г. в 07:14, vldekov ***@***.***>:

**Reply by vldekov** [reply]
ид таблицы «список контрагентов» совпадают с  ид таблицы «протокол
анализа заявки». Эта часть работает корректно. Ни как не получается заменить ИД в поле insert на ИМЯ по ИД с таблицы "список контрагентов".

**Comment by vldekov** [comment]
Есть решение! Достаточно было изменить формат поля insert на селект ( по таблице "список контрагентов" ). Спасибо!

**Comment by poluninatm** [comment]
Прошу обратить внимание, что последовательность ид и их непрерывность
тотумом не гарантируется - только уникальность.

В случае конкурирующих процессов это соответствие, если оно обеспечивается
логикой тотума, а не записью напрямую в БД, может быть нарушено.

Я бы не стала полагаться на соответствие, если не пишу в БД напрямую именно
его, а хранила в одной из таблиц ид соответствия второй.

Вт, 25 апр. 2023 г. в 08:00, vldekov ***@***.***>:
